src/components/ha-icon.ts

```diff
@@ -291,17 +101,16 @@ export class HaIcon extends LitElement {
 
     this._legacy = false;
 
-    if (iconName in mdiRenameMapping) {
-      iconName = mdiRenameMapping[iconName];
-      const message = `Icon ${iconPrefix}:${origIconName} was renamed to ${iconPrefix}:${iconName}, please change your config, it will be removed in version 0.115.`;
-      // eslint-disable-next-line no-console
-      console.warn(message);
-      fireEvent(this, "write_log", {
-        level: "warning",
-        message,
-      });
-    } else if (mdiRemovedIcons.has(iconName)) {
-      const message = `Icon ${this.icon} was removed from MDI, please replace this icon with an other icon in your config, it will be removed in version 0.115.`;
+    if (iconName in mdiDeprecatedIcons) {
+      const deprecatedIcon = mdiDeprecatedIcons[iconName];
+      let message: string;
+
+      if (deprecatedIcon.change === "renamed") {
+        message = `Icon ${iconPrefix}:${iconName} was renamed to ${iconPrefix}:${deprecatedIcon.newName}, please change your config, it will be removed in version ${deprecatedIcon.removeIn}.`;
+        iconName = deprecatedIcon.newName!;
+      } else {
+        message = `Icon ${iconPrefix}:${iconName} was removed from MDI, please replace this icon with an other icon in your config, it will be removed in version ${deprecatedIcon.removeIn}.`;
+      }
```

Comment:
```suggestion
      if (deprecatedIcon.removed) {
        message = `Icon ${iconPrefix}:${iconName} was removed from MDI, please replace this icon with an other icon in your config, it will be removed in version ${deprecatedIcon.removeIn}.`;
      } else if (deprecatedIcon.newName) {
        message = `Icon ${iconPrefix}:${iconName} was renamed to ${iconPrefix}:${deprecatedIcon.newName}, please change your config, it will be removed in version ${deprecatedIcon.removeIn}.`;
        iconName = deprecatedIcon.newName!;
      }
```